debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java

```diff
@@ -866,8 +866,9 @@ private void readTableSchema(final AtomicReference<String> sql, final JdbcConnec
         });
     }
 
-    private boolean shouldRecordTableSchema(final MySqlSchema schema, final Filters filters, TableId id) {
-        return !schema.isStoreOnlyMonitoredTablesDdl() || filters.tableFilter().test(id);
+    private boolean shouldRecordTableSchema(final MySqlSchema schema, final Filters filters, TableId id) throws SQLException {
+        return (!schema.isStoreOnlyMonitoredTablesDdl() || filters.tableFilter().test(id)) &&
+                !filters.ignoredTableFilter().test(id);
```

Comment:
This should be baked right into `tableFilter()` itself. I.e. how it's AND-ed with the built-in filter internally, this should be done similarly for the ignored tables.

Reply:
But since there's an `||` on `!schema.isStoreOnlyMonitoredTablesDdl()` it would be ignored if `table.ignore.builtin=false` right?

Reply:
No, it would not be ignored. The `||` operator doesn't short-circuit if the first operand returns false; it will evaluate the second operand then. I'm going to push an update to this PR in a second.